Use `getentropy()` instead of `/dev/urandom` on Emscripten

kleisauke opened the pull request:
`/dev/urandom` is usually available on Emscripten, except when using
the special `NODERAWFS` filesystem backend, which replaces all normal
filesystem access with direct Node.js operations.

Since this filesystem backend directly access the filesystem on the
OS, it is not recommended to depend on `/dev/urandom`, especially
when trying to run the Wasm binary on OSes that are not Unix-based.

This can be considered a non-functional change, since Emscripten
implements `/dev/urandom` in the same way as `getentropy()` when not
linking with `-sNODERAWFS`.

kleisauke committed:
Use getentropy() instead of /dev/urandom on Emscripten

`/dev/urandom` is usually available on Emscripten, except when using
the special `NODERAWFS` filesystem backend, which replaces all normal
filesystem access with direct Node.js operations.

Since this filesystem backend directly access the filesystem on the
OS, it is not recommended to depend on `/dev/urandom`, especially
when trying to run the Wasm binary on OSes that are not Unix-based.

This can be considered a non-functional change, since Emscripten
implements `/dev/urandom` in the same way as `getentropy()` when not
linking with `-sNODERAWFS`.

kleisauke committed:
Update libc dependency of std to 0.2.140

kleisauke commented:
This is ready for review now that libc 0.2.140 was released.

ChrisDenton commented:
lgtm

@bors r+

bors commented:
:pushpin: Commit 4d40cdfe2bed80f1fa7c1fe2a1526adb284d1c1d has been approved by `ChrisDenton`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=ChrisDenton 4d40cdfe2bed80f1fa7c1fe2a1526adb284d1c1d -->
<!-- homu: {"type":"Approved","sha":"4d40cdfe2bed80f1fa7c1fe2a1526adb284d1c1d","approver":"ChrisDenton","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->

bors commented:
:hourglass: Testing commit 4d40cdfe2bed80f1fa7c1fe2a1526adb284d1c1d with merge 8e6b904aeb2f7cd2dc46ea94a5390252d1303d9b...
<!-- homu: {"type":"BuildStarted","head_sha":"4d40cdfe2bed80f1fa7c1fe2a1526adb284d1c1d","merge_sha":"8e6b904aeb2f7cd2dc46ea94a5390252d1303d9b"} -->

bors commented:
:broken_heart: Test failed - [checks-actions](https://github.com/rust-lang-ci/rust/actions/runs/4548420657/jobs/8019699989)
<!-- homu: {"type":"BuildFailed","builder_url":"https://github.com/rust-lang-ci/rust/actions/runs/4548420657/jobs/8019699989","builder_name":"checks-actions"} -->

kleisauke committed:
Update Emscripten on CI to 2.0.5

`getentropy()` is available since Emscripten 2.0.5.

See:
https://github.com/emscripten-core/emscripten/pull/12240

kleisauke commented:
It looks like we need to bump the minimum supported Emscripten version to 2.0.5 for this (which is 2 months newer than 1.39.20), see commit 7b40eb71a90d93e825e32394d30f199a8492342d.

ChrisDenton commented:
Ah, ok. Let's try again...

@bors r+

bors commented:
:pushpin: Commit 7b40eb71a90d93e825e32394d30f199a8492342d has been approved by `ChrisDenton`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=ChrisDenton 7b40eb71a90d93e825e32394d30f199a8492342d -->
<!-- homu: {"type":"Approved","sha":"7b40eb71a90d93e825e32394d30f199a8492342d","approver":"ChrisDenton","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->

bors commented:
:hourglass: Testing commit 7b40eb71a90d93e825e32394d30f199a8492342d with merge 10338571cfa7c2be0aeed309141ced91454d41e1...
<!-- homu: {"type":"BuildStarted","head_sha":"7b40eb71a90d93e825e32394d30f199a8492342d","merge_sha":"10338571cfa7c2be0aeed309141ced91454d41e1"} -->

bors commented:
:sunny: Test successful - [checks-actions](https://github.com/rust-lang-ci/rust/actions/runs/4559262896/jobs/8044767793)
Approved by: ChrisDenton
Pushing 10338571cfa7c2be0aeed309141ced91454d41e1 to master...
<!-- homu: {"type":"BuildCompleted","approved_by":"ChrisDenton","base_ref":"master","builders":{"checks-actions":"https://github.com/rust-lang-ci/rust/actions/runs/4559262896/jobs/8044767793"},"merge_sha":"10338571cfa7c2be0aeed309141ced91454d41e1"} -->
